conans/test/README.md

```diff
@@ -0,0 +1,88 @@
+
+# Conan Testing
+
+Conan tests fall into three categories:
+
+- **Unit tests** in `conans/test/unittests` folder. These tests should test small pieces of code like
+  functions, methods, or properties. As long as it's possible they should not rely on anything
+  external like the file system or system configuration,  and in case they need to do it should be
+  mocked.
+
+- **Integration tests** in `conans/test/integration` folder. We consider integration tests the ones that
+  only will need pure python to execute, but that may test the interaction between different Conan
+  modules. They could test the result of the execution of one or several Conan commands, but shouldn't
+  depend on any external tools like compilers, build systems, or version-control system
+  tools.
+
+- **Functional tests** in `conans/test/functional` folder. Under this category, we add tests that are
+  testing the complete Conan functionality. They may call external tools (please read the section
+  below to check the tools installed on the CI). These tests should be avoided as long as
+  it's possible as they may take considerable CI time.
+
+## Writing tests
+
+We use [Pytest](https://docs.pytest.org/en/stable/) as the testing framework. There are some
+important things to have in mind regarding test discovery and style.
+
+### Naming files and methods
+
+Pytest follows this [convention](https://docs.pytest.org/en/stable/goodpractices.html) for test
+discovery:
+- Name your Python test files starting in `test_`.
+
+```
+test
+├── README.md
+├── conftest.py
+├── unittests
```

Comment:
```suggestion
├── integration
```

Related to `unittests`, which is the example here, but it was something I wanted to comment in this PR.

This is me **completely opinionated about this**: I REALLY like the tests under `unittests` to follow EXACTLY the same directory/files structure as the codebase, why?
 * no doubts where to add a test
 * no doubts where to check if a function is tested before writing one more test (duplicated).
 * when you are refactoring something, you only need to run one folder while changing the implementation.

We can talk about this later, but then I'd change the example here and show a file from `functional` or `integration`

Reply:
I agree, unittests should follow the same layout as the prod codebase.